pilot/pkg/config/kube/gateway/backend_policies.go

```diff
@@ -325,9 +338,16 @@ func BackendTrafficPolicyCollection(
 		if i.Spec.SessionPersistence != nil {
 			unsupported = append(unsupported, "sessionPersistence")
 		}
-		// TODO(https://github.com/istio/istio/issues/55838): implement i.Spec.RetryConstraint. This doesn't have upstream Envoy support
 		if i.Spec.RetryConstraint != nil {
-			unsupported = append(unsupported, "retryConstraint")
+			// TODO: add support for interval.
+			retryBudget = &networking.TrafficPolicy_RetryBudget{}
+			if i.Spec.RetryConstraint != nil {
+				retryBudget.Percent = &wrapperspb.DoubleValue{Value: float64(*i.Spec.RetryConstraint.Budget.Percent)}
```

Comment:
`i.Spec.RetryConstraint != nil` doesn't seem to imply `i.Spec.RetryConstraint.Budget.Percent != nil` so this may panic?